### DISCUSSION TITLE
Why does tqdm not work?

### DISCUSSION THREAD

**Discussion by mjspeck** [modular/modular#336] [Q&A] [answered]
The following code snippet using `tqdm` in Mojo doesn't work.

```python
from PythonInterface import Python

let tqdm = Python.import_module("tqdm")

for i in tqdm.tqdm(range(2)):
    print(i)
```

Running this in the `HelloMojo.ipynb` notebook provided by Mojo gives this error:

```
error: Expression [20]:23:23: invalid call to '__call__': argument #1 cannot be converted from '_ZeroStartingRange' to 'PythonObject'
    for i in tqdm.tqdm(range(2)):
             ~~~~~~~~~^~~~~~~~~~

/.modular/Kernels/mojo/Python/PythonObject.mojo:881:5: function declared here
    fn __call__(self, *args: PythonObject) raises -> PythonObject:
    ^

error: Expression [20]:24:15: use of unknown declaration 'i'
        print(i)
              ^
```

Why does this not work?

**Comment by strangemonad** [comment, marked as answer]
@mjspeck you'll want to do that in a `%%python` block for now. There's no way, at the moment, for mojo to know that the given `tqdm : PythonObject` implements a range iterator. The `for in` expression is currently hard coded to look for a `_ZeroStartingRange`.

This will at least need traits implemented but also sorting out how traits bridge with PythonObject protocols (related to #298)
